src/Univapay/Resources/ThreeDSMPI.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Univapay\Resources;
+
+use Univapay\Enums\Reason;
+use Univapay\Errors\UnivapayLogicError;
+
+class ThreeDSMPI
+{
+    public $authenticationValue;
+    public $eci;
+    public $dsTransactionId;
+    public $serverTransactionId;
+    public $messageVersion;
+    public $transactionStatus;
+
+    public function __construct(
+        $authenticationValue = null,
+        $eci = null,
+        $dsTransactionId = null,
+        $serverTransactionId = null,
+        $messageVersion = null,
+        $transactionStatus = null
+    ) {
```

Comment:
I think if you remove all the `= null` bits, the params all become required no? Then all the other validations becomes unnecessary.

Reply:
well indeed, what i'm thinking! thanks

Reply:
validation still necessary as user might put `empty` / `null` value